### ISSUE TITLE
Implementing the OneCycleLR scheduler ?

### ISSUE THREAD

**Issue by robinmonjo** [elixir-nx/polaris#7]
Pytorch has the one cycle LR scheduler: https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.OneCycleLR.html

It seems to me that implementing this kind of scheduler is not possible at the moment because scheduler in Polaris have an API that takes as input the step and returns a learning rate.

In the one cycle LR scenario, the scheduler should also be able to mutate the momentum of the optimizer (so affect the optimizer state).

Am I missing something ?
